### PR TITLE
Adding RabbitMQ deploy for kubernetes + prometheus integration

### PR DESCRIPTION
**Pull request by katcipis**
as mentioned on #1312

### CODE REVIEW

**Comment by beorn7** [comment]
@jimmidyson @fabxc I think you are both more qualified to comment on this than I am.

**Comment by jimmidyson** [comment]
The example YAMLs look fine.

@katcipis Could you add a README for how to apply & detail the important bits of the YAMLs so that other users can learn from them?

Whether the project wants a contrib dir like this is a decision for @fabxc, @beorn7, etc really. Seems useful to me.

**Comment by beorn7** [comment]
What's the rationale behind the `contrib`? Why not have it in `documentation/examples/kubernetes-rabbitmq/`? (I wasn't part of the discussion in https://github.com/prometheus/prometheus/issues/1312 .)

**Comment by katcipis** [comment]
@jimmidyson will do

**Comment by sdurrheimer** [comment]
> What's the rationale behind the contrib? Why not have it in documentation/examples/kubernetes-rabbitmq/? (I wasn't part of the discussion in #1312 .)

@beorn7 Maybe to differenciate examples maintained by Prometheus and examples provided by the community ?

**Comment by beorn7** [comment]
@sdurrheimer My gut feeling is that everything in the Prometheus org should be maintained somehow by people with push rights. Cf. 3rd party exporters, which are in a different org. If an example is too esoteric, it should be somewhere else, in a blog post of some kind or a separate GH org. If it is an example relevant for many users, it should be in the example directory without another level of nesting.

But I don't feel strongly. Let's wait what @fabxc thinks.

**Comment by brian-brazil** [comment]
I agree with @beorn7

**Comment by katcipis** [comment]
@jimmidyson I added a README.md to explain the example on details. Seems good enough ?

File: documentation/examples/contrib/kubernetes/rabbitmq/README.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+and on the [available examples](./documentation/examples).
+
+After you got Kubernetes service discovery up and running you just need to advertise that RabbitMQ
+is exposing metrics. To do that you need to define a service that:
+
+* Exposes the exporter port
+* Add the annotation: prometheus.io/scrape: "true"
+* Add the annotation: prometheus.io/port: "9090"
+
+And you should be able to see your RabbitMQ exporter being scrapped on prometheus status page.
```
Comment (jimmidyson) [inline documentation/examples/contrib/kubernetes/rabbitmq/README.md]: typo: scraped

File: documentation/examples/contrib/kubernetes/rabbitmq/README.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+Since the ip that will be scrapped will be the pod endpoint it is important that the node
```
Comment (jimmidyson) [inline documentation/examples/contrib/kubernetes/rabbitmq/README.md]: Typo: scraped

Comment (jimmidyson) [inline documentation/examples/contrib/kubernetes/rabbitmq/README.md]: And capitalise Prometheus please!

File: documentation/examples/contrib/kubernetes/rabbitmq/README.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+where prometheus is running have access to the Kubernetes overlay network
```
Comment (jimmidyson) [inline documentation/examples/contrib/kubernetes/rabbitmq/README.md]: Prometheus

**Comment by jimmidyson** [comment]
@katcipis Overall looks good - thanks!

**Comment by katcipis** [comment]
Typos fixed

**Comment by beorn7** [comment]
Let's then move it up into `documentation/examples/kubernetes-rabbitmq/...` rather than `documentation/examples/contrib/kubernetes/rabbitmq/...`

File: documentation/examples/contrib/kubernetes/rabbitmq/README.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
+# RabbitMQ Scraping
+
+This is an example on how to setup RabbitMQ so Prometheus can scrap data from it.
```
Comment (juliusv) [inline documentation/examples/contrib/kubernetes/rabbitmq/README.md]: scrap -> scrape

File: documentation/examples/contrib/kubernetes/rabbitmq/README.md [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+It uses a third party [RabbitMQ exporter](https://github.com/kbudde/rabbitmq_exporter).
+
+Since the [RabbitMQ exporter](https://github.com/kbudde/rabbitmq_exporter) needs to
+connect on RabbitMQ management API to scrap data, and it defaults to localhost, it is
```
Comment (juliusv) [inline documentation/examples/contrib/kubernetes/rabbitmq/README.md]: "connect to the"

"to scrape"

File: documentation/examples/contrib/kubernetes/rabbitmq/README.md [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+easier to simply embed the **kbudde/rabbitmq-exporter** on the same pod as RabbitMQ,
+this way they share the same network.
+
+With this pod running you will have the exporter scraping data, but Prometheus have not
```
Comment (juliusv) [inline documentation/examples/contrib/kubernetes/rabbitmq/README.md]: have -> has

File: documentation/examples/contrib/kubernetes/rabbitmq/README.md [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+yet found the exporter and is not scraping data from it.
+
+For more details on how to use kubernetes service discovery take a look on the
```
Comment (juliusv) [inline documentation/examples/contrib/kubernetes/rabbitmq/README.md]: kubernetes -> Kubernetes

Comment (juliusv) [inline documentation/examples/contrib/kubernetes/rabbitmq/README.md]: on -> at

File: documentation/examples/contrib/kubernetes/rabbitmq/README.md [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+connect on RabbitMQ management API to scrap data, and it defaults to localhost, it is
+easier to simply embed the **kbudde/rabbitmq-exporter** on the same pod as RabbitMQ,
+this way they share the same network.
+
+With this pod running you will have the exporter scraping data, but Prometheus have not
+yet found the exporter and is not scraping data from it.
+
+For more details on how to use kubernetes service discovery take a look on the 
+[documentation](http://prometheus.io/docs/operating/configuration/#kubernetes-sd-configurations-kubernetes_sd_config)
+and on the [available examples](./documentation/examples).
```
Comment (juliusv) [inline documentation/examples/contrib/kubernetes/rabbitmq/README.md]: on -> at

File: documentation/examples/contrib/kubernetes/rabbitmq/README.md [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+
+After you got Kubernetes service discovery up and running you just need to advertise that RabbitMQ
+is exposing metrics. To do that you need to define a service that:
+
+* Exposes the exporter port
+* Add the annotation: prometheus.io/scrape: "true"
```
Comment (juliusv) [inline documentation/examples/contrib/kubernetes/rabbitmq/README.md]: The structure here doesn't fit to "define a service that:". How about "Has the annotation: ..."? Same below.

File: documentation/examples/contrib/kubernetes/rabbitmq/README.md [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+* Add the annotation: prometheus.io/port: "9090"
+
+And you should be able to see your RabbitMQ exporter being scraped on Prometheus status page.
+Since the ip that will be scraped will be the pod endpoint it is important that the node
```
Comment (juliusv) [inline documentation/examples/contrib/kubernetes/rabbitmq/README.md]: s/ip/IP/

File: documentation/examples/contrib/kubernetes/rabbitmq/README.md [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+where Prometheus is running have access to the Kubernetes overlay network
+(flannel, weave, aws, or any of the other options that Kubernetes gives to you).
```
Comment (juliusv) [inline documentation/examples/contrib/kubernetes/rabbitmq/README.md]: weave -> Weave
aws -> AWS

(flannel is actually lower-cased)

File: documentation/examples/contrib/kubernetes/rabbitmq/README.md [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+and on the [available examples](./documentation/examples).
+
+After you got Kubernetes service discovery up and running you just need to advertise that RabbitMQ
+is exposing metrics. To do that you need to define a service that:
+
+* Exposes the exporter port
+* Add the annotation: prometheus.io/scrape: "true"
+* Add the annotation: prometheus.io/port: "9090"
+
+And you should be able to see your RabbitMQ exporter being scraped on Prometheus status page.
```
Comment (juliusv) [inline documentation/examples/contrib/kubernetes/rabbitmq/README.md]: on -> on the

File: documentation/examples/contrib/kubernetes/rabbitmq/README.md [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+Since the ip that will be scraped will be the pod endpoint it is important that the node
+where Prometheus is running have access to the Kubernetes overlay network
```
Comment (juliusv) [inline documentation/examples/contrib/kubernetes/rabbitmq/README.md]: have -> has

**Comment by katcipis** [comment]
Thanks @juliusv, tried to fix all the typos, take a look. Also moved to the suggested directory.

File: documentation/examples/kubernetes-rabbitmq/README.md [hunk 14]
```diff
@@ -0,0 +1,28 @@
+# RabbitMQ Scraping
+
+This is an example on how to setup RabbitMQ so Prometheus can scrape data from it.
+It uses a third party [RabbitMQ exporter](https://github.com/kbudde/rabbitmq_exporter).
+
+Since the [RabbitMQ exporter](https://github.com/kbudde/rabbitmq_exporter) needs to
+scrape the RabbitMQ management API to scrap data, and it defaults to localhost, it is
```
Comment (fabxc) [inline documentation/examples/kubernetes-rabbitmq/README.md]: s/scrap/scrape/

**Comment by fabxc** [comment]
:+1: thanks. Good for merge after final typo.

**Comment by juliusv** [comment]
Merging and will fix typo.

**Pull request merged**